### PR TITLE
Feature/153/implement finish of the stepper

### LINKED ISSUE #153: (SP: 3) Implement finish of the stepper
UserStory: #66, #67 
Blocked by: #152  

## Description
Implement sending all data to the server besides a photo by finishing

## Design 
Image
## Acceptance Criteria
- [x] When a user clicks finish it should send collected data from the stepper to user
- [x] When a user fills the fields into steps it should store this data for sending to the server
- [x] When a user clicks back it should show the previous step with filled fields that were filled by the user before

### PR DESCRIPTION
**Pull request by vldkk**
Implemented sending data from all stepper's parts to the server and stored data between steps before sending to the server. Covered all of this implementation with unit tests.
- [x] When a user clicks finish it should send collected data from the stepper to the user
- [x] When a user fills the fields into steps it should store this data for sending to the server
- [x] When a user clicks back it should show the previous step with filled fields that were filled by the user before

How it works:
https://www.loom.com/share/4ca1ec521cfa4c07960c25a1ea126c39?sid=80f80ce9-3a88-4c68-80f5-a4f203749222
Coverage with unit tests for GeneralInfoStep:
image
image_2024-03-01_16-04-15
Coverage with unit tests for SubjectsStep:
image
image_2024-03-01_15-33-21
Coverage with unit tests for LanguageStep:
image
image_2024-03-01_16-07-28

### CODE REVIEW

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx [hunk 1]
```diff
@@ -42,7 +46,15 @@ const GeneralInfoStep = ({ btnsBox }) => {
...
-
+  useEffect(() => {
+    handleStepData(stepLabel, {
+      firstName,
+      lastName,
+      country,
+      city,
+      professionalSummary
+    })
+  }, [firstName, lastName, country, city, professionalSummary])
```
Comment (iva-stasia) [inline src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx]: Please pay attention to the linter warning here.

File: src/containers/tutor-home-page/language-step/LanguageStep.jsx [hunk 2]
```diff
@@ -26,7 +28,9 @@ const LanguageStep = ({ btnsBox }) => {
   const handleClear = () => {
     setSelectedLanguage('')
   }
-
+  useEffect(() => {
+    handleStepData(stepLabel, selectedLanguage)
+  }, [selectedLanguage])
```
Comment (iva-stasia) [inline src/containers/tutor-home-page/language-step/LanguageStep.jsx]: Please pay attention to the linter warning here.

**Review by iva-stasia — APPROVED** [review]
Great job!
Please check the linter warnings.

File: src/containers/tutor-home-page/language-step/LanguageStep.jsx (hunk 2 above)
Comment (vldkk) [reply src/containers/tutor-home-page/language-step/LanguageStep.jsx]: I noticed it and checked, and everything is okay with these dependencies)

**Review by holus-bolus — APPROVED** [review]

**Review by iaremchuk — APPROVED** [review]

**Review by Wiret02 — APPROVED** [review]
Good work!

**Comment by OliaHrytsak** [comment]
I've just testing your code. Works well!
I'm waiting for pushing it for github!))

**Review by OliaHrytsak — APPROVED** [review]

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx (hunk 1 above)
Comment (vlad-khrychov) [reply src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx]: @vldkk but the error still hasn't gone away

File: src/containers/tutor-home-page/language-step/LanguageStep.jsx (hunk 2 above)
Comment (vlad-khrychov) [reply src/containers/tutor-home-page/language-step/LanguageStep.jsx]: No, it is not okay, `handleStepData` and `stepLabel` are external function for this effect, so they must be added as dependencies.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
There are a lot of mistakes with dependencies in `useEffect`s

**Review by vlad-khrychov — APPROVED** [review]
I hope it still works now. And you will not forget to create a new issue.

**Pull request merged**